benchmarks/asv/app_suite.py

```diff
@@ -134,3 +136,73 @@ def peakmem_minvertex_heavy(self):
         hyperparams = {'n_layers': self.n_layers,
                        'graph': self.graph}
         benchmark_qaoa(hyperparams)
+
+
+class ML_light:
+    """Benchmark a hybrid quantum-classical machine learning application with a small dataset."""
+
+    n_features = 4
+    n_samples = 20
+
+    def time_ml_autograd_light(self):
+        """Time 50 training steps of a hybrid quantum machine learning example in the autograd interface."""
+        hyperparams = {'n_layers': self.n_features,
+                       'n_samples': self.n_samples,
+                       'interface': 'autograd'}
+        benchmark_machine_learning(hyperparams)
+
+    def time_ml_torch_light(self):
+        """Time 50 training steps of a hybrid quantum machine learning example in the torch interface."""
+        hyperparams = {'n_layers': self.n_features,
+                       'n_samples': self.n_samples,
+                       'interface': 'torch'}
+        benchmark_machine_learning(hyperparams)
+
+    def time_ml_tf_light(self):
+        """Time 50 training steps of a hybrid quantum machine learning example in the tensorflow interface."""
+        hyperparams = {'n_layers': self.n_features,
+                       'n_samples': self.n_samples,
+                       'interface': 'tf'}
+        benchmark_machine_learning(hyperparams)
+
+    def peakmem_ml_autograd_light(self):
+        """Benchmark peak memory of 50 training steps of a hybrid quantum machine learning example
+        in the autograd interface."""
+        hyperparams = {'n_layers': self.n_features,
+                       'n_samples': self.n_samples}
+        benchmark_machine_learning(hyperparams)
+
+
+class ML_heavy:
+    """Benchmark a hybrid quantum-classical machine learning application with a large dataset."""
+
+    n_features = 10
+    n_samples = 100
```

Comment:
Takes about 5 min per function below.

Reply:
For the heavier functions, we should specify [benchmarking specifications](https://asv.readthedocs.io/en/stable/benchmarks.html#id5) like `repeat`, `number`, or `sample_time`.  We should run as may iterations of the heavy functions.

Reply:
Shall we do this in a separate PR to be consistent?

Reply:
Sounds good.